CMakeLists.txt

```diff
@@ -2,15 +2,22 @@ cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
 
 project(RigelEngine)
 
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 
 # Project settings
 ###############################################################################
 
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+      "Build type for Swift [Debug, RelWithDebInfo, Release, MinSizeRel]"
```

Comment:
Hah, took it from the Swift codebase (awesome quality by the way, recommend it).
Learned a lot from it.

Will fix

Reply:
About the build type, sure it can do without the description. But I'd still make it a cache variable, i.e. with this syntax:
`set(CMAKE_BUILD_TYPE "Release" CACHE STRING "" FORCE)` to ensure it persists.

Reply:
Sounds good! 👍🏻